bears/r/FormatRBear.py

```diff
@@ -69,24 +69,15 @@ def create_arguments(filename, file, config_file,
                 1
             ```
         """
-        rcode = ('library(formatR);'
-                 'formatR::tidy_source('
-                 'source="' + escape(filename, '"\\') + '",'
-                 'comment={r_keep_comments},'
-                 'blank={r_keep_blank_lines},'
-                 'arrow={r_use_arrows},'
-                 'brace.newline={r_braces_on_next_line},'
-                 'indent='
-                 '{tab_width}'.format(tab_width=tab_width,
-                                      r_keep_comments=_map_to_r_bool(
-                                          r_keep_comments),
-                                      r_keep_blank_lines=_map_to_r_bool(
-                                          r_keep_blank_lines),
-                                      r_use_arrows=_map_to_r_bool(
-                                          r_use_arrows),
-                                      r_braces_on_next_line=_map_to_r_bool(
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"',
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
+                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
+                   'comment=' + _map_to_r_bool(r_keep_comments),
+                   'arrow=' + _map_to_r_bool(r_use_arrows),
+                   'indent=' + str(tab_width)}
+        if r_max_expression_length != 0:
```

Comment:
ah okay :)
Done :)

2016-06-28 11:50 GMT+01:00 Mischa Krüger <notifications@github.com>:

> In bears/r/FormatRBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/525#discussion_r68735437>
> :
>
> > -                                      r_keep_blank_lines=_map_to_r_bool(
> > -                                          r_keep_blank_lines),
> > -                                      r_use_arrows=_map_to_r_bool(
> > -                                          r_use_arrows),
> > -                                      r_braces_on_next_line=_map_to_r_bool(
> > -                                          r_braces_on_next_line)))
> > -        # Disable r_max_expression_length if it is equal to 0
> > -        rcode += ('' if r_max_expression_length == 0
> > -                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
> > +        options = {'source="' + escape(filename, '"\\') + '"',
> > +                   'blank=' + _map_to_r_bool(r_keep_blank_lines),
> > +                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
> > +                   'comment=' + _map_to_r_bool(r_keep_comments),
> > +                   'arrow=' + _map_to_r_bool(r_use_arrows),
> > +                   'indent=' + str(tab_width)}
> > +        if r_max_expression_length != 0:
>
> @Adrianzatreanu <https://github.com/Adrianzatreanu> means to change
>
> if r_max_expression_length != 0:
>
> to
>
> if r_max_expression_length:
>
> as bool(some_int) is effectively this:
>
> def __bool__(some_int):
>     return some_int != 0
>
> :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/525/files/bfeb123dfdf4cb0a601d5b72ae92178a3ea8c61e#r68735437>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xot6q47I_IsuYz96N6z8wrPkm0UF_ks5qQPxogaJpZM4I14vI>
> .
>